collector/consumer/exporter/otelexporter/defaultadapter/net_adapter.go

```diff
@@ -0,0 +1,264 @@
+package defaultadapter
+
+import (
+	"github.com/Kindling-project/kindling/collector/model"
+	"github.com/Kindling-project/kindling/collector/model/constlabels"
+	"github.com/Kindling-project/kindling/collector/model/constnames"
+	"github.com/Kindling-project/kindling/collector/model/constvalues"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+type NetGaugeGroupAdapter struct {
+	*NetAdapterManager
+	*NetAdapterConfig
+}
+
+type NetAdapterConfig struct {
+	StoreTraceAsMetric bool
+	StoreTraceAsSpan   bool
+	StorePodDetail     bool
+	StoreExternalSrcIP bool
+}
+
+func (n *NetGaugeGroupAdapter) Adapt(gaugeGroup *model.GaugeGroup) ([]*AdaptedResult, error) {
+	if gaugeGroup.Name != constnames.AggregatedNetRequestGaugeGroup && gaugeGroup.Name != constnames.SingleNetRequestGaugeGroup {
+		// skip non net gaugeGroup
+		return nil, nil
+	}
+
+	if gaugeGroup.Name == constnames.AggregatedNetRequestGaugeGroup {
+		return n.dealWithPreAggGaugeGroups(gaugeGroup)
+	} else {
+		return n.dealWithSingleGaugeGroup(gaugeGroup)
+	}
```

Comment:
Could this be simplified using `switch-case`?

Reply:
Good idea